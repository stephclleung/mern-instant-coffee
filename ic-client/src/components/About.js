import React from 'react';
import { connect } from 'react-redux';
import { setLocation } from '../actions/location';
import { Container, Row, Col } from 'reactstrap';

export const About = (props) => {
    props.setLocation();
    return (
        <Container>
            <Row>
                <Col xs="auto">
                    <img src={"https://i.imgur.com/ebLwdYJ.png"} alt={"this is steph."} style={{ width: 100 + "%" }}></img>
                </Col>
            </Row>
            <Row>
                <Col xs="auto" >
                    <h3>Hi there!
                    Thanks for checking out my app.</h3>
                    I made this app as a practice for developing with MongoDB, Express, React and Node.
                    <p>Feel free to poke around.</p>
                    <p>If you haven't already done so, please also visit my portfolio at</p>
                    <div className="externalLinks"><center><a className="links" href="https://www.cl-leung.com" target="_blank" rel="noopener noreferrer"> www.cl-leung.com.</a></center></div>
                    <p>Hope you have a great day :).</p>
                </Col>
            </Row>
        </Container>
    )

}

const mapDispatchToProps = (dispatch) => {
    return {
        setLocation: () => dispatch(setLocation('about'))
    }
}

export default connect(undefined, mapDispatchToProps)(About);
//export default About;