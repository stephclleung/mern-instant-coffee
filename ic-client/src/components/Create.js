import React from 'react';
import { Link } from 'react-router-dom';
import InstantCoffeeForm from './InstantCoffeeForm';

//TODO : dispatch "add" when InstantCoffeeForm submits.
class Create extends React.Component {
   
    
    render() {
        return (
            <div>
                <InstantCoffeeForm />
                <Link to="/">Back</Link>
            </div>
        );
    }
};

export default Create;
