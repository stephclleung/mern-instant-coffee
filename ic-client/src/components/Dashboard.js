import React from 'react';
import InstantCoffeeList from './InstantCoffeeList';

const Dashboard = () =>{
    return (
        <div>
            {console.log("ICL: ", InstantCoffeeList)}
            <p>Dashboard</p>
            <InstantCoffeeList />
        </div>
    );
};

export default Dashboard;
