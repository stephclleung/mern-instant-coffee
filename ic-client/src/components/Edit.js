import React from 'react';
import InstantCoffeeForm from './InstantCoffeeForm';
import { connect } from 'react-redux';
import { removeInstantCoffeeFromDB, editInstantCoffeeToDB } from '../actions/instant-coffee';
import { handleOtherError } from '../actions/error';
import { setLocation } from '../actions/location';
import { Button } from 'reactstrap';


export class Edit extends React.Component {
    onRemoveCoffee = () => {
        this.props.removeInstantCoffeeFromDB(this.props.instantCoffee.id)
        this.props.history.push('/')
    }
    onEditCoffee = (coffeeUpdates) => {
        if (Object.entries(coffeeUpdates.changedFields).length !== 0) {
            try {
                this.props.editInstantCoffeeToDB(this.props.instantCoffee.id, coffeeUpdates);
            } catch (error) {
                this.props.handleOtherError();
                this.props.history.push('/')
            }
        }
        this.props.history.push('/')
    }
    render() {
        this.props.setLocation();
        return (
            <div>
                <Button onClick={this.onRemoveCoffee} className="col-lg-6 mb-3 btn-block align-item-middle">Remove Coffee</Button>
                <InstantCoffeeForm instantCoffee={this.props.instantCoffee} onSubmit={this.onEditCoffee} />
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        removeInstantCoffeeFromDB: (id) => dispatch(removeInstantCoffeeFromDB(id)),
        editInstantCoffeeToDB: (id, coffeeUpdates) => dispatch(editInstantCoffeeToDB(id, coffeeUpdates)),
        handleOtherError: () => dispatch(handleOtherError()),
        setLocation: () => dispatch(setLocation('edit'))
    }
}

const mapStateToProps = (state, props) => ({
    instantCoffee: state.instantCoffees.find((ic) => ic.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(Edit);
//export default Edit;