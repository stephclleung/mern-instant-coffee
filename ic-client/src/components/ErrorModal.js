import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { removeError } from '../actions/error'

class ErrorModal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            toggle: true
        }
    }
    onToggle = () => {
        //set toggle back false
        this.setState((prevState) => ({ toggle: !prevState }))
        //dispatch request to reset state to no error.
        this.props.removeError();
    }
    render() {
        return (
            <Modal
                isOpen={this.state.toggle}
                centered={true}
            >
                <ModalHeader
                    toggle={this.onToggle}
                >
                    Something went wrong.
                </ModalHeader>
                <ModalBody>
                    {this.props.errorMessage}
                </ModalBody>
                <ModalFooter>
                    <Button
                        onClick={this.onToggle}
                    >
                        Ok
                    </Button>
                </ModalFooter>
            </Modal>
        );
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        removeError: () => dispatch(removeError())
    }
}

export default connect(null, mapDispatchToProps)(ErrorModal);
// export default ErrorModal;