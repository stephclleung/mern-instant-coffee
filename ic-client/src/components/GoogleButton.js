
import { GoogleLogout, GoogleLogin } from 'react-google-login';
import React from 'react';

const GoogleButton = (props) => (
    <div>
        {console.log("Am butt", props)}
        {!!props.user.userName ? (
            <div>
                <p>Name: {props.user.userName}</p>
                <Logout onLogoutSuccess={props.onGoogleLogout} />
            </div>
        ) : (
                <Login onSuccess={props.onGoogleLogin} />
            )}
    </div>
);

const Login = (props) => (
    <div>
        {console.log("Am log in , ", props)}
        <GoogleLogin
            className="002"
            clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
            buttonText="Login"
            onRequest={loading}
            onSuccess={props.onSuccess}
            onFailure={tempFunction}
            cookiePolicy={'single_host_origin'}
        />
    </div>
)

const Logout = (props) => (
    <div>
        {console.log("Am log out, ", props)}
        {console.log("Am client ID", process.env.REACT_APP_GOOGLE_CLIENT_ID)}
        <GoogleLogout
            className="001"
            clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
            buttonText="Logout"
            onRequest={loading}
            onLogoutSuccess={props.onLogoutSuccess}
            onLogoutFailure={tempFunction}
        >
        </GoogleLogout>
    </div>
)

const tempFunction = (data) => {
    console.log("Temp function fired", data);
}

const loading = () => {
    console.log('loading') // eslint-disable-line
}
export default GoogleButton;