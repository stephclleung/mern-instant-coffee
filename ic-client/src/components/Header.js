import React from 'react';
import { NavLink } from 'react-router-dom';
import {
    Navbar,
    Collapse,
    Card,
    CardBody,
    Form,
    InputGroup,
    Input,
    Button,
    InputGroupButtonDropdown,
    Dropdown,
    DropdownToggle,
    DropdownMenu,
    DropdownItem
} from 'reactstrap';
import { setTextFilter } from '../actions/filter';
import { connect } from 'react-redux';

export class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            introCollapse: false,
            dropDownToggle: false,
            sortBy: "Sort By.."
        }
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortByChange = () => {
    }
    onDropDownMenuChange = (e) => {
        const sortBy = e.target.value;
        this.setState(() => ({ sortBy }))
    }
    onDropDownToggle = () => {
        this.setState((prevState) => ({ dropDownToggle: !prevState.dropDownToggle }))
    }
    onIntroToggle = () => {
        this.setState(prevState => ({ introCollapse: !prevState.introCollapse }))
    }
    render() {
        return (
            <div>
                <Navbar color="light" expand="md" fixed="top">
                    <div className="flex-md-column">
                        <h1 onClick={this.onIntroToggle}>Coffee...?</h1>
                        <Collapse isOpen={this.state.introCollapse}>
                            <Card >
                                <CardBody>
                                    Hi there!
                                </CardBody>
                            </Card>
                        </Collapse>
                    </div>
                    <Button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse">
                        <span className="navbar-toggler-icon"></span>
                    </Button>
                    <div className="collapse navbar-collapse w-100 flex-md-column" id="navbarCollapse">
                        <ul className="navbar-nav ml-auto small mb-2 mb-md-2">
                            <li><h4><NavLink to="/" activeClassName="is-active" exact={true} ><span role="img" aria-label="coffee">‚òï</span></NavLink></h4></li>
                            <li><h4><NavLink to="/create" activeClassName="is-active"><span role="img" aria-label="new">üìù</span></NavLink></h4></li>
                        </ul>
                        <Form className="form-inline ml-auto flex-md-row">
                            <InputGroup>
                                <Input
                                    type="text"
                                    className="form-control col-form-label-sm border-dark"
                                    placeholder="Find an instant coffee.."
                                    value={this.props.filter.text}
                                    onChange={this.onTextChange}
                                />
                                <InputGroupButtonDropdown
                                    addonType="append"
                                    toggle={this.onDropDownToggle}
                                    isOpen={this.state.dropDownToggle}
                                >
                                    <DropdownToggle
                                        caret
                                        size="sm"
                                        className="form-control col-form-label-sm border-dark flex-fixed"
                                    >
                                        {this.state.sortBy}
                                    </DropdownToggle>
                                    <DropdownMenu
                                        size="sm"
                                    >
                                        <DropdownItem value="Price" onClick={this.onDropDownMenuChange}>Price</DropdownItem>
                                        <DropdownItem value="Amount" onClick={this.onDropDownMenuChange}>Amount</DropdownItem>
                                        <DropdownItem value="Aroma" onClick={this.onDropDownMenuChange}>Aroma</DropdownItem>
                                        <DropdownItem value="Acidity" onClick={this.onDropDownMenuChange}>Acidity</DropdownItem>
                                        <DropdownItem divider />
                                        <DropdownItem value="Sort By.." onClick={this.onDropDownMenuChange}>...</DropdownItem>
                                    </DropdownMenu>
                                </InputGroupButtonDropdown>
                            </InputGroup>
                        </Form>
                    </div>
                </Navbar>
            </div>
        );
    }
};

const mapStateToProps = (state) => ({
    filter: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text))
})


export default connect(mapStateToProps, mapDispatchToProps)(Header);
//export default Header;
