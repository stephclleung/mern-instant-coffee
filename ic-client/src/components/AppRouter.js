import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import ErrorModal from './ErrorModal';
import Create from './Create';
import Dashboard from './Dashboard';
import NoMatch from './NoMatch';
import Header from './Header';
import Edit from './Edit';
import { Container } from 'reactstrap';
import { connect } from 'react-redux';

const AppRouter = (props) => {
    return (
        <BrowserRouter>
            <Header />
            <Container pt="4">
                <div className="m-3">
                    <Switch>
                        <Route exact={true} path="/" component={Dashboard} />
                        <Route path="/create" component={Create} />
                        <Route path="/login" />
                        <Route path="/edit/:id" component={Edit} />
                        <Route component={NoMatch} />
                    </Switch>
                </div>
            </Container>
            {props.errors.showErrorMessage &&
                <ErrorModal errorMessage={props.errors.errorMessage} showErrorMessage={props.errors.showErrorMessage} />}
        </BrowserRouter>
    );
};

const mapStateToProps = (state) => ({
    errors: state.errors
})

//export default AppRouter;
export default connect(mapStateToProps)(AppRouter);
