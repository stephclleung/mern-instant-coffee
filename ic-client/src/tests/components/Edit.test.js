import React from 'react';
import { Edit } from '../../components/Edit';

import { instantCoffee } from '../fixtures/instant-coffee-data';
import { shallow } from 'enzyme';

test("Should render Edit page correctly", () => {
    const wrapper = shallow(<Edit props={instantCoffee[0]} />)
    expect(wrapper).toMatchSnapshot();
});

test("Should redirect to dashboard after deleting", () => {
    let coffee = instantCoffee[0];
    coffee._id = "123ABCDEFG";
    let removeInstantCoffeeFromDB = jest.fn();    //Mock functions
    let history = { push: jest.fn() }
    const wrapper = shallow(
        <Edit
            instantCoffee={coffee}
            removeInstantCoffeeFromDB={removeInstantCoffeeFromDB}
            history={history}
        />)
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
})