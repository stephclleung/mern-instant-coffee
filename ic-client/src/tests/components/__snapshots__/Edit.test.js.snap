// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Edit page correctly 1`] = `ShallowWrapper {}`;
